### ISSUE TITLE
Allow construct IDs to use variables within `forEach`

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#122]
### Describe the bug

`for..of`, etc., variables are permitted for construct IDs within the looping process, but not within `forEach`. 

### Expected Behavior

The following codes do not result in an error.

```typescript
    ipSetRulesToCreate.forEach((ruleToCreate, i) => {
      const rule = new IpSetRule(this, ruleToCreate.ipSetNameSuffix, {  // use variables in construct ID
        scope: this.webAclScope,
        rulePriority,
        ...ruleToCreate,
      });
    });
```

### Current Behavior

The following codes result in an error.

```typescript
    ipSetRulesToCreate.forEach((ruleToCreate, i) => {
      const rule = new IpSetRule(this, ruleToCreate.ipSetNameSuffix, { // Error
        scope: this.webAclScope,
        rulePriority,
        ...ruleToCreate,
      });
    });
```

### eslint-cdk-plugin Version

1.1.1

### Node.js Version

22.11.0

### Language

TypeScript

### Checklist

- [x] I have searched for similar issues before creating this one
- [x] I have provided all the necessary information to reproduce the issue
- [x] I have verified this issue is related to eslint-cdk-plugin and not another dependency
